proposals/2020-10-28-Experimental.md

```diff
@@ -0,0 +1,128 @@
+
+- Operation
+- Parameter
+- Schema
+
+This specifies that the operation, parameter or schema is not yet stable and SHOULD be used with caution.
+
+### Operation Object
+
+...
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+... | ... | ...
+<a name="operationExperimental"></a>experimental | `boolean` | Specifies that an operation is in experimental status, meaning it may change outside of the normal breaking change process. Consumers SHOULD use with caution. Default value is `false`.
+
+### Parameter Object
+
+...
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+... | ... | ...
+<a name="parameterExperimental"></a>experimental | `boolean` | Specifies that a parameter is in experimental status, meaning it may change outside of the normal breaking change process. Consumers SHOULD use with caution. Default value is `false`. Cannot be `true` when the parameter is `required`.
+
+### Schema Object
+
+...
+
+##### Fixed Fields
+
+Field Name | Type | Description
+---|:---:|---
+... | ... | ...
+<a name="schemaExperimental"></a>experimental | `boolean` | Specifies that a schema is in experimental status, meaning it may change outside of the normal breaking change process. Consumers SHOULD use with caution. Default value is `false`.
+
+### Example Spec
+
+```yaml
+  /asset/constraints:
+    get:
+      tags:
+        - Asset
+        - Constraints
+      summary: Get a set of asset constraints
+      operationId: constraints
+      parameters:
+        - name: siteToken
+          in: query
+          description: Site token obtained from Site API
+          required: true
+          schema:
+            type: string
+      experimental: true
+```
+### Prior Art
+
+This kind of requirement is handled for TypeScript libraries by [api-extractor](https://api-extractor.com/pages/tsdoc/doc_comment_syntax/#release-tags) - they have both "alpha" and "beta" markers with a somewhat opinionated flow attached - I'm not sure that level of granularity is necessary. But the "beta" and "public" ones map well to the motivations described here:
+
+> - **beta**: Indicates that an API item has been released as a preview or for experimental purposes. Third parties are encouraged to try it and provide feedback. However, a “beta” API should NOT be used in production, because it may be changed or removed in a future version.
+> - **public**: Indicates that an API item has been officially released, and is now part of the supported contract for a package. If the SemVer versioning scheme is used, then the API signature cannot be changed without a MAJOR version increment.
+
+### Unanswered Questions
+
+- If an operation is not marked as experimental, but it is using a schema which is (i.e. as its request object), then it is implicitly also unstable. Would this usage be considered invalid?
+- Should `experimental` and `deprecated` be mutually exclusive?
```

Comment:
No, something can be experimental and become deprecated (for example, the experiment evolved or is abandoned).

Reply:
I tend to agree and will make this edit. Just want to call out that it could be seen as conflicting with the idea of having a single "lifecycle" enum as mentioned elsewhere.